Fix FortiGate Cluster Sync status

rudybroersma opened the pull request:
The revert in https://github.com/librenms/librenms/pull/16202 for https://github.com/librenms/librenms/pull/14921 didn't solve the issue. The bug was introduced before, but PR 14921 didn't fix it nonetheless.

This PR *does* fix it.

![image](https://github.com/user-attachments/assets/bd9b2c84-1bc8-4c93-be7d-da8c87fef2dc)

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [X] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [X] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

PipoCanaja commented on tests/data/fortigate_500d.json:
Hi @rudybroersma 
You should not remove a JSON, only recompute it.

rudybroersma replied:
I had removed that test data because the tests failed due to some (unrelated) OSPF differences in tests results. I'll add it back, please have a look

rudybroersma commented:
I had removed that test data because the tests failed due to some (unrelated) OSPF differences in tests results. I'll add it back, please have a look

PipoCanaja commented:
@rudybroersma Looks OK now indeed :D

PipoCanaja reviewed (approved):
LGTM

PipoCanaja commented on includes/discovery/sensors/state/fortigate.inc.php:
```diff
@@ -36,31 +36,51 @@
             'state',
             $device,
             $fgHaStatsSyncStatus_num . '.' . $index,
-            $fgHaStatsSyncStatus_txt . '.' . $index,
```
Any reason you removed $fgHaStatsSyncStatus_txt ?

rudybroersma replied:
Yes, if I put that back the state indexes no longer work and the sensors get the status 'OK' again (instead of In Sync or Out of Sync). So it breaks the state index translations.

Additionally if I look at the PHP example at the bottom of https://docs.librenms.org/Developing/Sensor-State-Support/#advanced-example it shows:

discover_sensor($valid['sensor'], 'state', $device, $num_oid.$index, $index, $state_name,

In that example parameter 5 is also just the index, not the text representation of the OID plus index.
Using just the index for parameter 5 seems to be the way to go according to the documentation?

murrant commented on includes/discovery/sensors/state/fortigate.inc.php:
```diff
@@ -19,8 +19,8 @@
  *
- * @copyright  2020 Net Entertainment AB
- * @author     Patrik Jonsson <patrik.jonsson@gmail.com>
+ * @copyright  2020 Net Entertainment AB, 2024 CTNET BV
```
It actually should be additional @author and @copyright lines :)

rudybroersma replied:
Fixed!

murrant commented:
Question, so is this a single sensor for the entire cluster, where before it was a sensor for each node in the cluster?

rudybroersma commented:
> Question, so is this a single sensor for the entire cluster, where before it was a sensor for each node in the cluster?

No, I didn't change anything to the number of sensors. I only fixed the state translation.
There is still one sensor per cluster node.

I added the explanation in the code because earlier versions of this file (prior to 2022) had an additional check as can be seen here:
https://github.com/librenms/librenms/blob/591384f321a891325eb785f7732179318ef91096/includes/discovery/sensors/state/fortigate.inc.php

This patch removed that:
https://github.com/librenms/librenms/blob/28a5811923c74b1e04140576e23f0ec6502948aa/includes/discovery/sensors/state/fortigate.inc.php

But, that last patch also broke the state translation.

I chose not to bring back the cluster state sensor with a 0 / 1 value, because it's not 100% correct.
FortiGate clusters can consist of a maximum of 4 nodes, and that sensor from PR #12300 only works properly with a 2-node cluster. If you have a 4-node cluster and 1 or 2 nodes are lost, the sensor value doesn't change (because the number of nodes in the cluster is still higher than 1)

In PR #13980 that sensor was changed to become a simple sensor that displays the number of nodes. So for a 2-node cluster it'll be 2 and for a 4-node cluster it'll be 4. And subsequently for a single-node cluster (which is a valid config according to FortiNet) the value will be 1. The administrator can then add an alert rule to trigger if the sensor has changed to a value below the actual number of cluster members.

Unfortunately there is no way to determine the expected number of nodes from the fortigate. FortiGate will silently remove any cluster member from the configuration when it goes offline

PR #13980 also seems to have broken the state index translations. I fixed that.
